crates/virtio-queue/src/lib.rs

```diff
@@ -1544,6 +1642,11 @@ mod tests {
         q.state.next_used = Wrapping(0);
         assert_eq!(q.needs_notification().unwrap(), true);
         assert_eq!(q.needs_notification().unwrap(), false);
+
+        m.write_obj::<u16>(u16::MAX - 3, avail_addr.unchecked_add(4 + qsize as u64 * 2))
```

Comment:
Can you add a comment here to explain why you're calling write again & needs notification is true?

Reply:
done